### PR TITLE
refactor!: organize and export more utils

### PR DESCRIPTION
**Pull request by ssoorriiin**
BREAKING CHANGES:
1. "lib/features/modeling/util/ModelingUtil.js"
  - file removed
  - "getParent" function moved under "lib/util/DiUtil.js"
  - "isAny" function moved under "lib/util/ModelingUtil.js"

2. "lib/features/replace/BpmnReplace.js"
  - removed "hasEventDefinition" function and using already existing from "lib/util/DiUtil.js"

3. "lib/features/rules/BpmnRules.js"
  - removed "hasEventDefinition" function and using already existing from "lib/util/DiUtil.js"
  - functions moved under "lib/util/DiUtil.js": isDroppableBoundaryEvent, isBoundaryEvent, isBoundaryCandidate, hasNoEventDefinition, hasCommonBoundaryIntermediateEventDefinition, hasOneOfEventDefinitions

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
Hi @ssoorriiin, first thanks for your contribution.

Can you explain overall what do you want to achieve with this refactoring. As you already described, this would mean a major breaking change for our modeling toolkit and all applications above it. Let's understand the big benefit this would give.

**Comment by ssoorriiin** [comment]
Hi @pinussilvestrus ,

It all started with a personal need of importing `isBoundaryCandidate` (in a custom rules module), function that is dependent on couple more. Instead of copy/paste-ing too much code on a separate project, I thought it would be nice to actually just import it from `bpmn-js`.

Besides the need, it exports other functions that may be useful to import by others, and also consolidates utility functions instead of spreading them in multiple places.

Thank you!

File: lib/util/DiUtil.js [hunk 1]
```diff
@@ -59,3 +70,70 @@ export function hasEscalationEventDefinition(element) {
...
+    return true;
+  }
+
+  return (
+    is(element, 'bpmn:IntermediateCatchEvent') &&
+    hasCommonBoundaryIntermediateEventDefinition(element)
+  );
+}
+
+export function isBoundaryEvent(element) {
```
Comment (philippfromme) [inline lib/util/DiUtil.js]: How is this related to the DI?

Comment (ssoorriiin) [reply lib/util/DiUtil.js]: It's not 👍. I think a better place is under [lib/util/ModelUtil.js](https://github.com/bpmn-io/bpmn-js/pull/1215/files/3a709634185db6e85631c7badb716a6831517cf9?file-filters%5B%5D=.js&hide-deleted-files=true#diff-9fb402b8a60ec6b0c96714019a5756b7).

**Comment by philippfromme** [comment]
I don't think we're going to merge this PR. Neither does it add any functionality nor fix anything. What do you guys think? @barmac @nikku @oguzeroglu

**Comment by pinussilvestrus** [comment]
I also think it doesn't add that much benefits to the codebase.

**Comment by oguzeroglu** [comment]
@philippfromme +1

**Comment by philippfromme** [comment]
Closing this PR for now.

**Comment by nikku** [comment]
I guess what we could do is ensure that important BPMN utilities are exported.

Some day in the future we could also collect those into a single helper. Would work now, as tree-shaking is a thing.

**Comment by philippfromme** [comment]
> Some day in the future

Yes.
